### PR TITLE
PYIC-3201 Create CIMIT stub lambda dynamo db table

### PR DESCRIPTION
**Pull request by TalatAydinGDS**
### What changed

Create CIMIT stub lambda dynamo db table

### Why did it change

This stores CI and mitigation data for a given user id

### Issue tracking

- [PYI-3201](https://govukverify.atlassian.net/browse/PYI-3201)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: di-ipv-cimit-stub/deploy/template.yaml [hunk 1]
```diff
@@ -71,6 +71,65 @@ Resources:
...
+    Properties:
+      # checkov:skip=CKV_AWS_28: Point in time recovery is not necessary for this table.
+      TableName: !Sub "cimit-stub-${Environment}"
+      BillingMode: "PAY_PER_REQUEST"
+      AttributeDefinitions:
+        - AttributeName: "userId"
+          AttributeType: "S"
+        - AttributeName: "contraIndicatorCode"
+          AttributeType: "S"
+      KeySchema:
```
Comment (therealslimsladey) [inline di-ipv-cimit-stub/deploy/template.yaml:85]: The primary key is a composite key of `userId` and `contraIndicatorCode` with the userId being the partition (HASH) key and the contraIndicatorCode being the sort (RANGE) key, akin to the user-credentials-v2 and cri-response tables in core-back. No need for secondary indexes methinks.

So
```
      KeySchema:
        - AttributeName: "userId"
          KeyType: "HASH"
        - AttributeName: "contraIndicatorCode"
          KeyType: "RANGE"
```

File: di-ipv-cimit-stub/deploy/template.yaml [hunk 2]
```diff
@@ -71,6 +71,65 @@ Resources:
...
+        - AttributeName: "userId"
+          KeyType: "HASH"
+      GlobalSecondaryIndexes:
```
Comment (therealslimsladey) [inline di-ipv-cimit-stub/deploy/template.yaml]: As above, I don't think we need this.

File: di-ipv-cimit-stub/lib/build.gradle [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+	testImplementation "org.junit.jupiter:junit-jupiter:5.9.3",
+			"org.mockito:mockito-junit-jupiter:5.4.0"
+
+	compileOnly "org.projectlombok:lombok:1.18.22"
+	annotationProcessor "org.projectlombok:lombok:1.18.22"
+}
+
+
+sourceCompatibility = 17
```
Comment (therealslimsladey) [inline di-ipv-cimit-stub/lib/build.gradle]: we should be specifying both source and target compatibility and I think we're dropping to 11?

File: di-ipv-cimit-stub/lib/src/main/java/uk/gov/di/ipv/core/library/persistence/DataStore.java
Comment (therealslimsladey) [inline di-ipv-cimit-stub/lib/src/main/java/uk/gov/di/ipv/core/library/persistence/DataStore.java:1]: Are we really going to use the local running facility and all the methods that are present? Let's tidy this up once we know which methods we're going to be using in the management API

File: di-ipv-cimit-stub/lib/src/main/java/uk/gov/di/ipv/core/library/persistence/items/CimitStubItem.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    private Instant issuanceDate;
+    private long ttl;
+
+    private List<String> mitigations;
+
+    @DynamoDbPartitionKey
+    public String getUserId() {
+        return userId;
+    }
+    @DynamoDbSecondaryPartitionKey(indexNames = "contraIndicatorCode")
```
Comment (therealslimsladey) [inline di-ipv-cimit-stub/lib/src/main/java/uk/gov/di/ipv/core/library/persistence/items/CimitStubItem.java]: This is going to be the DynamoDbSortKey

**Review by therealslimsladey — CHANGES_REQUESTED** [review]

File: di-ipv-cimit-stub/deploy/template.yaml (hunk 1 above)
Comment (TalatAydinGDS) [reply di-ipv-cimit-stub/deploy/template.yaml:85]: Done

File: di-ipv-cimit-stub/deploy/template.yaml (hunk 2 above)
Comment (TalatAydinGDS) [reply di-ipv-cimit-stub/deploy/template.yaml]: Done

File: di-ipv-cimit-stub/lib/build.gradle (hunk 3 above)
Comment (TalatAydinGDS) [reply di-ipv-cimit-stub/lib/build.gradle]: Done

File: di-ipv-cimit-stub/lib/src/main/java/uk/gov/di/ipv/core/library/persistence/items/CimitStubItem.java (hunk 4 above)
Comment (TalatAydinGDS) [reply di-ipv-cimit-stub/lib/src/main/java/uk/gov/di/ipv/core/library/persistence/items/CimitStubItem.java]: Done

**Review by therealslimsladey — APPROVED** [review]
lgtm

**Pull request merged**